.github/workflows/ci-pr.yml

```diff
@@ -0,0 +1,61 @@
+name: CI-PR
+on:
+  pull_request:
+    branches: [master]
+concurrency:
+  group: ci-pr-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  ci-pr:
+    name: CI-PR
+    needs: [test, msrv]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Done
+        run: exit 0
+  test:
+    name: Tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        rust: [stable]
+        flags: ["--no-default-features --features 'std cargo'", "--features 'wrap_help yaml regex'"]
```

Comment:
We should name this flag as `features` and call them as I did in `ci.yml` to maintain consistency.

The naming them of here is important (instead of just providing the whole flag here) because it would improve the naming of the CI statuses and checks.

Reply:
If we want to improve the display for `ci-pr.yml`, my recommendation would be distinct jobs with their own matrices for each case.

I find `ci.yml` complicated and do not consider consistency with it a prerequiste.  Because Github Actions does not have a rich way for us to associate a label with a set of flags, we have 4 mutually exclusive core steps in the `test` job and people have to map what "test" means under each scenario.  As a contributor, I have found the "pretty" job names does not help me in figuring out what is going on (most of the time, its been cut off).  As someone modifying the pipeline, it feels brittle and hard to follow.

To be clear, I understand, at least one reason, why `ci.yml` needs to be so complicated: because of the complicated matrix, there is a lot of logic that would have to be duplicated if we split it out into separate jobs (the `exclude`s and target-specific setup).  However, `ci-pr.yml` does not have nearly as complicated as a matrix.

Reply:
I think the ability of naming the flags `all` or `none` definitely helps lower the bar to contribution. Sure, it's not a high priority, but I think if it's possible to keep that aspect in this PR without much problem I'd prefer it as well. 

Then we can move towards a better overall system in the future with a more concerted effort. Especially because we definitely will need some kind of system in place if (once) we get to a more `tokio`-esque modularity with many different flags.

Reply:
I've applied my middle ground option of distinct matrix jobs; let's see what we think of it.

Longer term, I wonder if we should look at a command runner, like `just` or `make`.  That way we can pass features in as a parameter and have that dispatch it out, rather than complicating the pipeline.

With that said, some caveats
- We should ensure people still get Github Actions time reporting and not include sequencing of steps
- We should make sure the tool echos the command being run so people can easily run it locally if they don't have the command runner locally or to give context to help them understand why it might have failed